plugin/src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/rest/RestPPLQueryAction.java

```diff
@@ -196,7 +196,9 @@ public void onFailure(Exception e) {
     Format format = pplRequest.format();
     ResponseFormatter<QueryResult> formatter;
     if (format.equals(Format.CSV)) {
-      formatter = new CsvResponseFormatter(pplRequest.sanitize());
+      formatter = new FlatResponseFormatter(",", pplRequest.sanitize());
+    } else if (format.equals(Format.RAW)) {
+      formatter = new FlatResponseFormatter("|", pplRequest.sanitize());
```

Comment:
Could you check if raw data need sanitize? Should it be either always false or at least false by default? Because I think it would be a surprise to users if we sanitize raw data by default.

Reply:
Feel that comma, tab, colon, or vertical bar(pipe) are just all column delimiter, that give me believe that sanitize should be applied to raw data, but need to confirm from document. Tested in excel that sanitize is implemented, but could not test raw format. Let me search around more.

1. https://en.wikipedia.org/wiki/Delimiter-separated_values
2. https://docs.microsoft.com/en-us/sql/integration-services/import-export-data/configure-flat-file-destination-sql-server-import-and-export-wizard?view=sql-server-ver15

Reply:
For RAW format: character '\' is put before the '|' character, as same as csv I think
=============Data Base===============
mysql> select * from tutorials_tbl;
+-------------+----------------+-----------------+-----------------+
| tutorial_id | tutorial_title | tutorial_author | submission_date |
+-------------+----------------+-----------------+-----------------+
|           1 | Learn PHP      | John Poul       | 2021-01-25      |
|           2 | Learn ,,,,PHP  | John||| Poul    | 2021-01-25      |
+-------------+----------------+-----------------+-----------------+
2 rows in set (0.00 sec)

=============export to text file limited by vertical bar ===============
mysql> SELECT * FROM tutorials_tbl
    -> INTO OUTFILE '/tmp/cancelled_orders2.csv' 
    -> FIELDS ENCLOSED BY '|' 
    -> LINES TERMINATED BY '\r\n';
Query OK, 2 rows affected (0.00 sec)

=============\ is put before the | character in field ===============
f8ffc257e573:tmp harowang$ sudo more  cancelled_orders2.csv
|1|     |Learn PHP|     |John Poul|     |2021-01-25|
|2|     |Learn ,,,,PHP| |John\|\|\| Poul|       |2021-01-25|

============= \ is put before the , character in field ===============
f8ffc257e573:tmp harowang$ sudo more  cancelled_orders3.csv
Password:
,1,     ,Learn PHP,     ,John Poul,     ,2021-01-25,
,2,     ,Learn \,\,\,\,PHP,     ,John||| Poul,  ,2021-01-25,